andela/team-odd-bn-backend
# #169258641 super admin should create user roles

### What does this PR do?
This PR is going to help super admin to assign roles to different users.

#### Description of Task to be completed?
**Why**
Different user roles are supported within the application with each role allowed different level of access

**User Roles**

Super Administrator - [IT Staff]
Travel Administrator
Travel Team Member
Manager
Requester
As a super administrator
I should be able to assign roles to any staff member (via their email address) based on the user roles listed above.

#### How should this be manually tested?
- `git clone https://github.com/andela/team-odd-bn-backend.git`
- `cd team-odd-bn-backend`
- git checkout -b ft-assign-user-role-169258641
- `npm install`
- update `.env` file with super admin credentials
- Drop existing migration `npm run dropTables`
- Update migration `npm run db-migrate`
- Populate the seed data `npm run db-seed-dev`
- `npm run dev-start` check API in development 
- `npm test` testing API

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#169258641](https://www.pivotaltracker.com/story/show/169258641)

#### Screenshots (if appropriate)
API Development 
<img width="1439" alt="Screen Shot 2019-11-18 at 14 26 34" src="https://user-images.githubusercontent.com/56690376/69052550-f89c4700-0a0f-11ea-957c-cef73b5c23b6.png">

API Documentation
<img width="1440" alt="Screen Shot 2019-11-19 at 21 31 08" src="https://user-images.githubusercontent.com/56690376/69179386-1e127900-0b14-11ea-9279-d7fb5028a316.png">

Available roles
<img width="1440" alt="Screen Shot 2019-11-21 at 12 08 20" src="https://user-images.githubusercontent.com/56690376/69343745-d6f0c900-0c76-11ea-9074-f64db159b226.png">

Available users
<img width="1440" alt="Screen Shot 2019-11-22 at 06 39 47" src="https://user-images.githubusercontent.com/56690376/69399181-de0aec00-0cf5-11ea-91ed-54c8555f289e.png">


#### Questions:
